src/ranch_acceptors_sup.erl

```diff
@@ -25,28 +25,49 @@ start_link(Ref, NumAcceptors, Transport) ->
 
 init([Ref, NumAcceptors, Transport]) ->
 	TransOpts = ranch_server:get_transport_options(Ref),
-	Logger = maps:get(logger, TransOpts, error_logger),
+	NumListenSockets = maps:get(num_listen_sockets, TransOpts, 1),
 	SocketOpts = maps:get(socket_opts, TransOpts, []),
-	%% We temporarily put the logger in the process dictionary
-	%% so that it can be used from ranch:filter_options. The
-	%% interface as it currently is does not allow passing it
-	%% down otherwise.
+	Logger = maps:get(logger, TransOpts, error_logger),
 	put(logger, Logger),
-	LSocket = case Transport:listen(SocketOpts) of
+	LSockets = start_listen_sockets(Ref, NumListenSockets, Transport, SocketOpts, Logger),
+	erase(logger),
+	[{_, BaseSocket}|_] = LSockets,
+	{ok, Addr} = Transport:sockname(BaseSocket),
+	ranch_server:set_addr(Ref, Addr),
+	Procs = [begin
+		LSocketId = (AcceptorId rem NumListenSockets) +1,
+		{_, LSocket} = lists:keyfind(LSocketId, 1, LSockets),
+		{
+			{acceptor, self(), AcceptorId},
+			{ranch_acceptor, start_link, [Ref, AcceptorId, LSocket, Transport, Logger]},
+			permanent, brutal_kill, worker, []
+		}
+	end || AcceptorId <- lists:seq(1, NumAcceptors)],
+	{ok, {{one_for_one, 1, 5}, Procs}}.
+
+-spec start_listen_sockets(any(), pos_integer(), module(), list(), module()) ->
+	[{pos_integer(), inet:socket()}].
+start_listen_sockets(Ref, NumListenSockets, Transport, SocketOpts, Logger) when NumListenSockets > 0 ->
+	case lists:keyfind(port, 1, SocketOpts) of
+		{port, Port} when Port > 0 ->
+			[{N, start_listen_socket(Ref, Transport, SocketOpts, Logger)} ||
+				N <- lists:seq(1, NumListenSockets)];
+		_ ->
+			BaseSocket = start_listen_socket(Ref, Transport, SocketOpts, Logger),
+			{ok, {_, Port}} = Transport:sockname(BaseSocket),
+			SocketOpts1 = [{port, Port}|lists:keydelete(port, 1, SocketOpts)],
+			[{1, BaseSocket}|[{N, start_listen_socket(Ref, Transport, SocketOpts1, Logger)} ||
+				N <- lists:seq(2, NumListenSockets)]]
```

Comment:
Do the lists comprehension outside of the case, there's no need to have two lists comprehensions doing the same thing.